### ISSUE TITLE
Signature not printed to stdout as per docs

### ISSUE THREAD

**Issue by budgester** [sigstore/cosign#2611]
**Description**

When running cosign and choosing not to upload signature the base encoded signature is not printed to stdout

Documentation specifies the base64 encoded signature should be printed

https://github.com/sigstore/cosign/blob/main/USAGE.md#sign-but-skip-upload-to-store-somewhere-else

**Version**

```
budgester@thor ~ $ ~/go/bin/cosign version
  ______   ______        _______. __    _______ .__   __.
 /      | /  __  \      /       ||  |  /  _____||  \ |  |
|  ,----'|  |  |  |    |   (----`|  | |  |  __  |   \|  |
|  |     |  |  |  |     \   \    |  | |  | |_ | |  . `  |
|  `----.|  `--'  | .----)   |   |  | |  |__| | |  |\   |
 \______| \______/  |_______/    |__|  \______| |__| \__|
cosign: A tool for Container Signing, Verification and Storage in an OCI registry.

GitVersion:    v1.13.1
GitCommit:     unknown
GitTreeState:  unknown
BuildDate:     unknown
GoVersion:     go1.18.1
Compiler:      gc
Platform:      linux/amd64
```

upload=false
```
budgester@thor ~ $ ~/go/bin/cosign sign --key cosign.key --upload=false 343327909574.dkr.ecr.eu-west-2.amazonaws.com/budgester@sha256:709288e86514674efc6d7f1d821d194a53f73733ae9fae31c634821922f6c4fa
Enter password for private key: 
```

Sucessful upload to AWS ECR
```
budgester@thor ~ $ ~/go/bin/cosign sign --key cosign.key 343327909574.dkr.ecr.eu-west-2.amazonaws.com/budgester@sha256:709288e86514674efc6d7f1d821d194a53f73733ae9fae31c634821922f6c4fa
Enter password for private key: 
Pushing signature to: 343327909574.dkr.ecr.eu-west-2.amazonaws.com/budgester
```

**Comment by znewman01** [comment]
Docs are wrong here. Thanks for noting and filing an issue; we'll fix the docs.

You want `--output-signature <filename>` . Is there any reason you'd need it output directly to STDOUT?

**Comment by budgester** [comment]
No reason for having it send to stdout, this is perfect was just following the docs.

My initial stumbling point was not understanding that cosign signs containers once they are in a repo rather i.e. after a push rather than straight after a build

Thanks for your support

**Comment by znewman01** [comment]
Reopening so that we remember to fix the docs here 😄

**Comment by gosixtyfive** [comment]
@znewman01 Looking to make my first contribution and would like to do the docs update for this issue.  Is the right next step to create an issue in sigstore/docs that references this one then do a PR against that one in docs before closing this?

Change would be something like this I believe:
```
-The base64 encoded signature is printed to stdout.
-This can be stored somewhere else.
+Write the base64 encoded signature to an output file
 
-$ cosign sign --key key.pem --upload=false user/demo
-Qr883oPOj0dj82PZ0d9mQ2lrdM0lbyLSXUkjt6ejrxtHxwe7bU6Gr27Sysgk1jagf1htO/gvkkg71oJiwWryCQ==
+$ cosign sign --key key.pem --upload=false --output-signature demo.sig user/demo
 ```

**Comment by budgester** [comment]
@znewman01 might be worth noting in some way that using 'cosign sign --help'  weget the options available
```
--output-certificate string  write the certificate to FILE
--output-signature string  write the signature to FILE
```

And also what --upload=false does
```
--upload  whether to upload the signature (default true)
```

The combination of using both --upload and --output-signature gives a nice way of testing and understanding what `cosign sign` creates without adding thing to a repo.

**Comment by gosixtyfive** [comment]
@znewman01 this now ok to close.  Resolved by sigstore/docs/pull/58

thx for review and guidance.

**Comment by znewman01** [comment]
Thanks for the fix! Hope to see you around the project 😄
